pkg/rbac/rbac_suite_test.go

```diff
@@ -0,0 +1,254 @@
+  name: view-red-metrics
+---
+`
+
+const systemMetricsAccessOnAllClusterYaml = `
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: view-system-metrics
+rules:
+  - apiGroups:
+      - "cluster.open-cluster-management.io"
+    resources:
+      - managedclusters
+    verbs:
+      - metrics/kube-system
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: view-system-metrics-binding
+subjects:
+  - kind: Group
+    apiGroup: rbac.authorization.k8s.io
+    name: system-admins
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: view-system-metrics
+---
+`
+
+var (
+	baseK8sConfig *rest.Config
+	baseK8sClient kubernetes.Interface
+	testEnv       *envtest.Environment
+	ctx           context.Context
+	cancel        context.CancelFunc
+	testUsers     = map[string]struct {
+		KubeClient kubernetes.Interface
+		Groups     []string
+	}{
+		"user-blue": {
+			nil, // set after startUp
+			[]string{"blue-admins"},
+		},
+		"user-red": {
+			nil, // set after startUp
+			[]string{"red-admins"},
+		},
+		"user-purple": {
+			nil, // set after startUp
+			[]string{"blue-admins", "red-admins"},
+		},
+		"user-no-specific-access": {
+			nil, // set after startUp
+			[]string{"no-specific-access"},
+		},
+		"user-sysadmin": {
+			nil, // set after startUp
+			[]string{"system-admins"},
+		},
+	}
+	testRbacResourceYamls = []string{blueMetricsAccessYaml, redMetricsAccessYaml, systemMetricsAccessOnAllClusterYaml}
+)
+
+func TestMain(m *testing.M) {
+	os.Exit(testMain(m))
+}
+
+func testMain(m *testing.M) int {
+	defer tearDown()
+
+	setUp()
+
+	return m.Run()
+}
+
+func setUp() {
+	testEnv = &envtest.Environment{}
+
+	var err error
+
+	baseK8sConfig, _ = testEnv.Start()
+
+	baseK8sClient, err = kubernetes.NewForConfig(baseK8sConfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	ctx, cancel = context.WithCancel(context.TODO())
+
+	// SetUp Users and their Groups
+	for user, userConfig := range testUsers {
+		userKubeClient, err := SetupUser(user, userConfig.Groups)
+		if err != nil {
+			panic(err.Error())
+		}
+
+		userConfig.KubeClient = userKubeClient
+		testUsers[user] = userConfig
+	}
+
+	// Setup RBAC resources for MetricsAccess
+	for _, testRbacResourceYaml := range testRbacResourceYamls {
+		err = SetupRbacResources(testRbacResourceYaml)
+		if err != nil {
+			panic(err.Error())
+		}
+	}
+}
+
+func tearDown() {
+	cancel()
+
+	err := testEnv.Stop()
+	if err != nil {
+		panic(err.Error())
+	}
+}
+
+func SetupUser(username string, groups []string) (kubernetes.Interface, error) {
+	user := envtest.User{
+		Name:   username,
+		Groups: groups,
+	}
+
+	testUser, err := testEnv.AddUser(user, baseK8sConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	testUserKClient, err := kubernetes.NewForConfig(testUser.Config())
+	if err != nil {
+		return nil, err
+	}
+
+	return testUserKClient, nil
+}
+
+func SetupRbacResources(resourcesYaml string) error {
```

Comment:
To follow Go naming conventions of capitalizing well known acronyms.
```suggestion
func SetupRBACResources(resourcesYaml string) error {
```